utils.py

```diff
@@ -243,22 +243,33 @@ def _split_lines_from_chunks(chunks):
 
 
 def resolve_ip(host):
-    cached_ip = ip_cache.get(host, None)
+    cached_ip = ip_cache.get().get(host, None)
     if cached_ip:
         return cached_ip
     else:
         try:
             logger.debug("trying to resolve ip address for host %s", host)
             ip = socket.gethostbyname(host)
-            ip_cache.set(host, ip)
+            ip_cache.get().set(host, ip)
             return ip
         except socket.gaierror:
             return None
-ip_cache = LRUCache()
 
 
-def set_ip_cache(new_ip_cache):
-    ip_cache = new_ip_cache
+class LRUCacheSingleton(object):
```

Comment:
This was really just the *easiest* solution I could think of. Definitely not the *best*.